packages/swing-store/src/transcriptStore.js

```diff
@@ -137,25 +164,25 @@ export function makeTranscriptStore(db, ensureTxn, noteExport) {
    */
   function initTranscript(vatID) {
     ensureTxn();
-    sqlWriteSpan.run(vatID, 0, 0, initialHash, 1, 0);
+    sqlWriteSpan.run(vatID, 0, 0, initialHash, 1);
   }
 
   const sqlGetCurrentSpanBounds = db.prepare(`
-    SELECT startPos, endPos, size, hash
+    SELECT startPos, endPos, hash
     FROM transcriptSpans
-    WHERE vatID = ? AND current = 1
+    WHERE vatID = ? AND isCurrent = 1
   `);
 
   /**
    * Obtain the bounds and other metadata for a vat's current transcript span.
    *
    * @param {string} vatID  The vat in question
    *
-   * @returns {{startPos: number, endPos: number, size: number, hash: string}}
+   * @returns {{startPos: number, endPos: number, hash: string}}
    */
   function getCurrentSpanBounds(vatID) {
     const bounds = sqlGetCurrentSpanBounds.get(vatID);
```

Comment:
Using `NULL` for `isCurrent` instead of `0`, and setting a `UNIQUE(vatID, isCurrent)` constraint should work.

Reply:
ooh, so the UNIQUE constraint allows an arbitrary number of NULLs, but only one row of any other value? Man I learn something new every day.

Reply:
Apparently. To be fair, I just learned that too ;)

Reply:
> only one row of any other value

To clarify, only one row of a given value. `NULL` is just considered to all be different from each other, like `NaN != NaN` in JS.

Reply:
So in SQL `NULL` means `NaD` ("not a datum")?  They could have just said that.  But I guess the kinds of people who invented relational databases were not the kinds of people who went in for humor, since they worked for IBM at the time.

I will attempt putting in the constraint.  It'll blow up right quick if you're wrong (which I don't think you are).

Reply:
Seems to work.  The hardest part actually was the shift to encoding false as `NULL` instead of `0`.